spec/support/test/stripe_mock_helper_spec.rb

```diff
@@ -18,12 +18,14 @@
   describe "#start" do
     it 'is safely reentrant' do
       StripeMockHelper.mock do
-        # create a plan 
-        StripeMockHelper.stripe_helper.create_plan(id: 'test_str_plan', amount:0, currency: 'usd', interval: 'year', name: 'test PLan')
+        # products are now required in plans as of Stripe gem 5.0
+        product = StripeMockHelper.stripe_helper.create_product
+        # create a plan
+        StripeMockHelper.stripe_helper.create_plan(id: 'test_str_plan', amount: 0, currency: 'usd', interval: 'year', name: 'test PLan', product: product.id)
```

Comment:
@caseyhelbling this is a follow up to https://github.com/CommitChange/houdini/pull/1041#discussion_r2008024426:

We don't need to create StripePlans or products for live code. Those would be managed entirely through Stripe itself. StripeMock allows us to create sample plans and products for testing so all we're doing here is proving that we can do.